### PR TITLE
MOTECH-2944 Add the ability to auto complete based on available field values in action

### PR DESCRIPTION
**Pull request by wstrzelczyk**

### CODE REVIEW

**Comment by pkornowski** [comment]
Actually it's not working for the key-value map. Please fix this.

**Comment by pkornowski** [comment]
@kmadej  Looks good to me.

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 1]
```diff
@@ -607,6 +695,97 @@
...
+                    return prepareField(fieldToPrepare, fullName);
+                };
+
+                scope.refreshListItems = function (event) {
+                    scope.filteredItems = [];
+                    if(event.keyCode === 8) {event.key = "";scope.rangySelection.anchorOffset = scope.rangySelection.anchorOffset -1;}
+                    if (scope.rangySelection.anchorNode && scope.rangySelection.anchorNode.nodeValue) {
+                        var value = scope.rangySelection.anchorNode.nodeValue;
+                        scope.searchText = value.substring(findStartIndex(value.trimRight(), scope.rangySelection.anchorOffset), scope.rangySelection.anchorOffset) + event.key;//extractText(element.html() + event.key, scope.debug);
+                        scope.filteredItems = getFilteredItems(items, scope.searchText.trim());
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: remove comments

File: modules/tasks/tasks/src/main/resources/webapp/partials/form-action.html [hunk 2]
```diff
@@ -71,7 +71,7 @@
             <div class="form-group" ng-show="selectedAction[$index].description">
                 <label class="col-md-2 col-sm-3 control-label">{{msg(selectedAction.description)}}</label>
             </div>
-            <div ng-if="selectedAction[$index]">
+            <div ui-if="selectedAction[$index]">
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/partials/form-action.html]: why do you change to ui-if ?

Comment (wstrzelczyk) [inline modules/tasks/tasks/src/main/resources/webapp/partials/form-action.html]: I just restored unnecessary previous changes

**Pull request merged**